openlane/steps/step.py

```diff
@@ -116,6 +119,55 @@ class StepSignalled(StepException):
 MetricsUpdate = Dict[str, Any]
 
 
+class ProcessStatsThread(Thread):
+    def __init__(self, process, interval=0.1):
+        Thread.__init__(
+            self,
+        )
+        self.process = process
+        self.result = None
+        self.interval = interval
+        self.props = {
+            "cpu": 0.0,
+            "cpu_time_user": 0.0,
+            "cpu_time_system": 0.0,
+            "cpu_time_iowait": 0.0,
+            "memory_rss": 0.0,
+            "memory_vms": 0.0,
+            "threads": 0.0,
```

Comment:
These can all be stored as just normal Python properties and then a serialization method can create a dictionary (i.e. `to_dict`)